lib/puppet/ssl/certificate_authority.rb

```diff
@@ -288,14 +288,15 @@ def sign(hostname, options={})
       cert_type = :ca
       issuer = csr.content
     else
-      options[:allow_dns_alt_names] = true if hostname == Puppet[:certname].downcase
       unless csr = Puppet::SSL::CertificateRequest.indirection.find(hostname)
         raise ArgumentError, "Could not find certificate request for #{hostname}"
       end
 
       cert_type = :server
       issuer = host.certificate.content
 
+      # This allows for various bootstrapping scenarios
```

Comment:
Can the `Puppet::SSL::CertificateRequest.indirection.find(hostname)` call change the options hash? Otherwise I'm in the same boat as @nathanielksmith and am not sure how this helps.